classes/order/OrderHistory.php

```diff
@@ -99,6 +99,8 @@ public function changeIdOrderState($new_order_state, $id_order, $use_existing_pa
         // executes hook
         if (in_array($new_os->id, array(Configuration::get('PS_OS_PAYMENT'), Configuration::get('PS_OS_WS_PAYMENT')))) {
             Hook::exec('actionPaymentConfirmation', array('id_order' => (int)$order->id), null, false, true, false, $order->id_shop);
+            //initialize invoice date
+            $order->invoice_date =date('Y-m-d H:i:s');
```

Comment:
@vincentbz it will be not accounted as invoiced cause there's no invoice_id. And how the merchants will get problems if we use this solution?